### PR TITLE
webgpu: Fix a bug in softmax

### PR DESCRIPTION
**Pull request by qjia7**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7607)

### CODE REVIEW

**Comment by qjia7** [comment]
@hujiajie Please take a look, thanks.

**Review by gyagp — APPROVED** [review]

**Review by xhcao — APPROVED** [review]
